generator/src/main/java/com/scottlogic/deg/generator/inputs/validation/ProfileFieldUntypedValidator.java

```diff
@@ -0,0 +1,89 @@
+package com.scottlogic.deg.generator.inputs.validation;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.constraints.atomic.AtomicConstraint;
+import com.scottlogic.deg.generator.constraints.atomic.IsInSetConstraint;
+import com.scottlogic.deg.generator.constraints.atomic.IsNullConstraint;
+import com.scottlogic.deg.generator.constraints.atomic.IsOfTypeConstraint;
+import com.scottlogic.deg.generator.decisiontree.*;
+import com.scottlogic.deg.generator.inputs.validation.messages.StringValidationMessage;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Rejects a profile if there are any fields that aren't positively assigned at least one type.
+ *
+ * This is complicated by conditional logic/alternation. The gold standard is to only allow a profile
+ * if there are no ways to select logical paths that result in a field not having determinable type(s).
+ * In reality this is computationally challenging, so this class employs a heuristic approach. It will
+ * reject something like:
+ *
+ *   x number
+ *   if (x equalTo 2) then y string
+ *   x equalTo 2
+ *
+ * But at least accepts semi-complicated cases like:
+ *
+ *   anyOf
+ *      x number
+ *      x string
+ *
+ * Crucially, it should never permit an invalid case.
+ *
+ * see {@link https://github.com/ScottLogic/datahelix/issues/767 #767} for more details
+ */
+public class ProfileFieldUntypedValidator implements ProfileValidator {
```

Comment:
Should probably have something to do with type mandatory in the name of the class.
e.g. MandatoryTypeValidator

Reply:
Typed renamed